From 68d4d67269e4b19727931ce746f3185d4a52100e Mon Sep 17 00:00:00 2001
From: Adrien Nader <adrien@notk.org>
Date: Sun, 6 Jan 2013 14:42:07 +0100
Subject: [PATCH 07/34] configure: for cross, a matching ocamlrun must be on
 the system.

---
 INSTALL   |  9 +++++++++
 configure | 20 ++++++++++++++++++++
 2 files changed, 29 insertions(+)

diff --git a/INSTALL b/INSTALL
index 98dfd31..b7ffc58 100644
--- a/INSTALL
+++ b/INSTALL
@@ -76,6 +76,15 @@ The "configure" script accepts the following options:
         hand. The installation instructions for gcc or emacs contain a
         complete list of configuration names.
 
+-target <targettype>            (default: same as -host)
+        The type of the target machine, in GNU's "configuration name"
+        format (CPU-COMPANY-SYSTEM or CPU-COMPANY-KERNEL-SYSTEM).
+        Setting this will setup OCaml as a cross-compiler which runs on
+        $host and produces code for $target. This requires a C toolchain
+        which also produces code for $target and a native OCaml
+        compiler of the exact same version (if you want a cross 4.00.1,
+        you need a native 4.00.1).
+
 -x11include <include_dir>       (default: determined automatically)
 -x11lib     <lib_dir>           (default: determined automatically)
         Location of the X11 include directory (e.g. /usr/X11R6/include)
diff --git a/configure b/configure
index ae110c7..b211dea 100755
--- a/configure
+++ b/configure
@@ -373,6 +373,26 @@ case $? in
          "Make sure the C compiler $cc is properly installed.";;
 esac
 
+# Determine which ocamlrun executable to use; for cross-compilation, a native
+# "ocamlrun" executable must be available on the system.
+if test x"$target" != x"$host"; then
+  if ! sh ./searchpath ocamlrun; then
+    err "Cross-compilation requires an ocaml runtime environment (the ocamlrun binary). Moreover, its version must be the same as the one you're trying to build (`cut -f1 -d+ < ../../VERSION`)."
+  else
+    ocaml_system_version=`ocamlrun -version | sed 's/[^0-9]*\([0-9.]\+\).*/\1/'`
+    ocaml_source_version=`sed -n '1 s/\([0-9\.]\+\).*/\1/ p' < ../../VERSION`
+    if test x"$ocaml_system_version" != x"$ocaml_source_version"; then
+      err "While you have an ocaml runtime environment, its version ($ocaml_system_version) doesn't match the version of these sources ($ocaml_source_version)."
+    else
+      CAMLRUN="ocamlrun"
+    fi
+  fi
+else
+  CAMLRUN=`cd ../.. && pwd`/boot/ocamlrun
+fi
+
+echo "CAMLRUN=$CAMLRUN" >> Makefile
+
 # Check the sizes of data types
 
 inf "Checking the sizes of integers and pointers..."
-- 
1.8.1.4

