From 0d2783db9d10ce894860b0a022281e49b8309842 Mon Sep 17 00:00:00 2001
From: Adrien Nader <adrien@notk.org>
Date: Tue, 25 Dec 2012 22:54:07 +0100
Subject: [PATCH 08/34] configure: add --target and use "target" instead of
 "host" for variables

---
 configure | 46 ++++++++++++++++++++++++++++------------------
 1 file changed, 28 insertions(+), 18 deletions(-)

diff --git a/configure b/configure
index b211dea..7f36d59 100755
--- a/configure
+++ b/configure
@@ -20,6 +20,7 @@ libdir=''
 mandir=''
 manext=1
 host_type=unknown
+target_type=""
 ccoption=''
 asoption=''
 asppoption=''
@@ -95,6 +96,8 @@ while : ; do
         shift;;
     -host*|--host*)
         host_type=$2; shift;;
+    -target*|--target*)
+        target_type=$2; shift;;
     -cc*)
         ccoption="$2"; shift;;
     -as)
@@ -207,7 +210,14 @@ fi
 if host=`../gnu/config.sub $host_type`; then :; else
   err "Please specify the correct host type with the -host option"
 fi
-inf "Configuring for a $host ..."
+inf "Configuring for host $host ..."
+
+if test -n "$target_type"; then
+  target="$target_type"
+else
+  target="$host"
+fi
+inf "Configuring for target $target ..."
 
 # Do we have gcc?
 
@@ -226,7 +236,7 @@ fi
 
 buggycc="no"
 
-case "$host,$cc" in
+case "$target,$cc" in
   i[3456]86-*-*,gcc*)
     case `$cc --version` in
       2.7.2.1) cat <<'EOF'
@@ -277,7 +287,7 @@ ostype="Unix"
 exe=""
 iflexdir=""
 
-case "$bytecc,$host" in
+case "$bytecc,$target" in
   cc,*-*-nextstep*)
     # GNU C extensions disabled, but __GNUC__ still defined!
     bytecccompopts="-fno-defer-pop $gcc_warnings -U__GNUC__ -posix"
@@ -475,7 +485,7 @@ esac
 
 # Determine alignment constraints
 
-case "$host" in
+case "$target" in
   sparc*-*-*|hppa*-*-*|arm*-*-*|mips*-*-*)
     # On Sparc V9 with certain versions of gcc, determination of double
     # alignment is not reliable (PR#1521), hence force it.
@@ -507,7 +517,7 @@ case "$host" in
 esac
 
 if $int64_native; then
-  case "$host" in
+  case "$target" in
     # PR#5088: autodetection is unreliable on ARM.  PR#5280: also on MIPS.
     sparc*-*-*|hppa*-*-*|arm*-*-*|mips*-*-*)
       if test $2 = 8; then
@@ -555,7 +565,7 @@ mksharedlibrpath=''
 natdynlinkopts=""
 
 if test $withsharedlibs = "yes"; then
-  case "$host" in
+  case "$target" in
     *-*-cygwin*)
       mksharedlib="$flexlink"
       mkmaindll="$flexlink -maindll"
@@ -649,7 +659,7 @@ fi
 natdynlink=false
 
 if test $withsharedlibs = "yes"; then
-  case "$host" in
+  case "$target" in
     *-*-cygwin*)                  natdynlink=true;;
     i[3456]86-*-linux*)           natdynlink=true;;
     i[3456]86-*-gnu*)             natdynlink=true;;
@@ -688,7 +698,7 @@ arch=none
 model=default
 system=unknown
 
-case "$host" in
+case "$target" in
   sparc*-*-solaris2.*)          arch=sparc; system=solaris;;
   sparc*-*-*bsd*)               arch=sparc; system=bsd;;
   sparc*-*-linux*)              arch=sparc; system=linux;;
@@ -731,7 +741,7 @@ case "$host" in
 esac
 
 # Some platforms exist both in 32-bit and 64-bit variants, not distinguished
-# by $host.  Turn off native code compilation on platforms where 64-bit mode
+# by $target.  Turn off native code compilation on platforms where 64-bit mode
 # is not supported.  (PR#4441)
 
 if $arch64; then
@@ -751,7 +761,7 @@ nativecccompopts=''
 nativecclinkopts=''
 nativeccrpath="$byteccrpath"
 
-case "$arch,$nativecc,$system,$host_type" in
+case "$arch,$nativecc,$system,$target" in
   *,*,nextstep,*)      nativecccompopts="$gcc_warnings -U__GNUC__ -posix"
                        nativecclinkopts="-posix";;
   *,*,rhapsody,*darwin[1-5].*)
@@ -839,7 +849,7 @@ echo "ARCMD=ar" >> Makefile
 
 if (SHELL=/bin/sh; export SHELL; (./sharpbang || ./sharpbang2) >/dev/null); then
   inf "#! appears to work in shell scripts."
-  case "$host" in
+  case "$target" in
     *-*-sunos*|*-*-unicos*)
       wrn "We won't use it, though, because under SunOS and Unicos it breaks" \
           "on pathnames longer than 30 characters"
@@ -1109,7 +1119,7 @@ if sh ./hasgot mktime; then
   echo "#define HAS_MKTIME" >> s.h
 fi
 
-case "$host" in
+case "$target" in
   *-*-cygwin*) ;;  # setsid emulation under Cygwin breaks the debugger
   *) if sh ./hasgot setsid; then
        inf "setsid() found."
@@ -1215,7 +1225,7 @@ echo "BNG_ASM_LEVEL=$bng_asm_level" >> Makefile
 systhread_support=false
 
 if test "$pthread_wanted" = "yes"; then
-  case "$host" in
+  case "$target" in
     *-*-solaris*)  pthread_link="-lpthread -lposix4"
                    pthread_caml_link="-cclib -lpthread -cclib -lposix4";;
     *-*-freebsd*)  pthread_link="-pthread"
@@ -1231,7 +1241,7 @@ if test "$pthread_wanted" = "yes"; then
     otherlibraries="$otherlibraries systhreads"
     bytecccompopts="$bytecccompopts -D_REENTRANT"
     nativecccompopts="$nativecccompopts -D_REENTRANT"
-    case "$host" in
+    case "$target" in
       *-*-freebsd*)
           bytecccompopts="$bytecccompopts -D_THREAD_SAFE"
           nativecccompopts="$nativecccompopts -D_THREAD_SAFE";;
@@ -1400,7 +1410,7 @@ if test "$x11_include" = "not found"; then
         x11_link="-lX11"
       else
         x11_libs="-L$dir"
-        case "$host" in
+        case "$target" in
           *-kfreebsd*-gnu) x11_link="-L$dir -lX11";;
           *-*-*bsd*) x11_link="-R$dir -L$dir -lX11";;
           *) x11_link="-L$dir -lX11";;
@@ -1527,7 +1537,7 @@ if test $has_tk = true; then
         -ltk$tkmaj$tkmin -ltcl$tclmaj$tclmin -lpthread $tkauxlibs \
                    Tcl_DoOneEvent
   then
-    case "$host" in
+    case "$target" in
       *-*-*bsd*) tk_libs="-R/usr/pkg/lib -L/usr/pkg/lib $tk_libs $tk_x11_libs -ltk$tkmaj$tkmin -ltcl$tclmaj$tclmin -lpthread $tkauxlibs";;
       *) tk_libs="-L/usr/pkg/lib $tk_libs $tk_x11_libs -ltk$tkmaj$tkmin -ltcl$tclmaj$tclmin -lpthread $tkauxlibs";;
     esac
@@ -1541,14 +1551,14 @@ if test $has_tk = true; then
   if sh ./hasgot $tk_libs $tk_x11_libs $tkauxlibs Tk_SetGrid; then
     inf "Tcl/Tk libraries found."
   elif sh ./hasgot -L/sw/lib $tk_libs $tk_x11_libs $tkauxlibs Tk_SetGrid; then
-    case "$host" in
+    case "$target" in
       *-*-*bsd*) tk_libs="-R/sw/lib -L/sw/lib $tk_libs";;
       *) tk_libs="-L/sw/lib $tk_libs";;
     esac
     inf "Tcl/Tk libraries found."
   elif sh ./hasgot -L/usr/pkg/lib $tk_libs $tk_x11_libs $tkauxlibs \
                    Tk_SetGrid; then
-    case "$host" in
+    case "$target" in
       *-*-*bsd*) tk_libs="-R/usr/pkg/lib -L/usr/pkg/lib $tk_libs";;
       *) tk_libs="-L/usr/pkg/lib $tk_libs";;
     esac
-- 
1.8.1.4

