From eeddee977792b41f8ada2281e3d7472458734fbc Mon Sep 17 00:00:00 2001
From: Adrien Nader <adrien@notk.org>
Date: Sun, 6 Jan 2013 19:19:03 +0100
Subject: [PATCH 17/34] ocamldoc: chose the "ocamldoc" command in configure.

---
 configure         | 58 ++++++++++++++++++++++++++++++++++++-------------------
 debugger/Makefile |  2 +-
 ocamldoc/Makefile | 27 ++++++++++++--------------
 3 files changed, 51 insertions(+), 36 deletions(-)

diff --git a/configure b/configure
index edcd9ac..a4bf5b8 100755
--- a/configure
+++ b/configure
@@ -436,26 +436,6 @@ case $? in
      fi;;
 esac
 
-# Determine which ocamlrun executable to use; for cross-compilation, a native
-# "ocamlrun" executable must be available on the system.
-if test x"$target" != x"$host"; then
-  if ! sh ./searchpath ocamlrun; then
-    err "Cross-compilation requires an ocaml runtime environment (the ocamlrun binary). Moreover, its version must be the same as the one you're trying to build (`cut -f1 -d+ < ../../VERSION`)."
-  else
-    ocaml_system_version=`ocamlrun -version | sed 's/[^0-9]*\([0-9.]\+\).*/\1/'`
-    ocaml_source_version=`sed -n '1 s/\([0-9\.]\+\).*/\1/ p' < ../../VERSION`
-    if test x"$ocaml_system_version" != x"$ocaml_source_version"; then
-      err "While you have an ocaml runtime environment, its version ($ocaml_system_version) doesn't match the version of these sources ($ocaml_source_version)."
-    else
-      CAMLRUN="ocamlrun"
-    fi
-  fi
-else
-  CAMLRUN=`cd ../.. && pwd`/boot/ocamlrun
-fi
-
-echo "CAMLRUN=$CAMLRUN" >> Makefile
-
 # Check the sizes of data types
 # OCaml needs a 32 or 64bit architectue and a 32-bit integer type.
 
@@ -1728,6 +1708,44 @@ case "$buggycc" in
     nativecccompopts="$nativecccompopts -fomit-frame-pointer";;
 esac
 
+# Determine which ocamlrun executable to use; for cross-compilation, a native
+# "ocamlrun" executable must be available on the system.
+rootdir=`cd ../.. && pwd`
+
+if test x"$target" != x"$host"; then
+  if ! sh ./searchpath ocamlrun; then
+    err "Cross-compilation requires an ocaml runtime environment (the ocamlrun binary). Moreover, its version must be the same as the one you're trying to build (`cut -f1 -d+ < ../../VERSION`)."
+  else
+    # Make sure the version of the system ocaml and the version of these
+    # sources match. Don't compare "+devXX" on purpose as it would be too
+    # annoying during development and people using SVN are expected to know
+    # what they're doing and how to handle basic issues.
+    ocaml_system_version=`ocamlc -version | cut -f1 -d+`
+    ocaml_source_version=`sed -n '1 s/\([0-9\.]\+\).*/\1/ p' < ../../VERSION`
+    if test x"$ocaml_system_version" != x"$ocaml_source_version"; then
+      err "The system OCaml toolchain is $ocaml_system_version but these sources are $ocaml_source_version.\nBuilding a cross-compiler requires versions to match."
+    else
+      camlrun="ocamlrun"
+      # Assert the host's ocamldoc supports shared libraries
+      ocamldoc="ocamlrun ocamldoc"
+    fi
+  fi
+else
+  camlrun=$rootdir/boot/ocamlrun
+  if $shared_libraries_supported; then
+    ocamldoc="\$(CAMLRUN) -I $rootdir/otherlibs/$unixlib -I $rootdir/otherlibs/str $rootdir/ocamldoc/ocamldoc"
+  else
+    ocamldoc="$rootdir/ocamldoc/ocamldoc"
+  fi
+fi
+
+cat >> Makefile << EOF
+ROOTDIR=$rootdir
+CAMLRUN=$camlrun
+OCAMLRUN=\$(CAMLRUN)
+OCAMLDOC=$ocamldoc
+EOF
+
 # Finish generated files
 
 cclibs="$cclibs $mathlib"
diff --git a/debugger/Makefile b/debugger/Makefile
index cf0fffb..75f4aa8 100644
--- a/debugger/Makefile
+++ b/debugger/Makefile
@@ -10,5 +10,5 @@
 #                                                                       #
 #########################################################################
 
-UNIXDIR=../otherlibs/unix
+UNIXDIR=../otherlibs/$(UNIXLIB)
 include Makefile.shared
diff --git a/ocamldoc/Makefile b/ocamldoc/Makefile
index 122f7e6..f78a4ff 100644
--- a/ocamldoc/Makefile
+++ b/ocamldoc/Makefile
@@ -14,7 +14,6 @@ include ../config/Makefile
 
 # Various commands and dir
 ##########################
-CAMLRUN=../boot/ocamlrun
 OCAMLC   = ../ocamlcomp.sh
 OCAMLOPT = ../ocamlcompopt.sh
 OCAMLDEP = $(CAMLRUN) ../tools/ocamldep
@@ -29,7 +28,6 @@ OCAMLPP=-pp './remove_DEBUG'
 ##############
 MKDIR=mkdir -p
 CP=cp -f
-OCAMLDOC_RUN=sh ./runocamldoc $(SUPPORTS_SHARED_LIBRARIES)
 OCAMLDOC_LIBCMA=odoc_info.cma
 OCAMLDOC_LIBCMI=odoc_info.cmi
 OCAMLDOC_LIBCMXA=odoc_info.cmxa
@@ -223,8 +221,7 @@ $(OCAMLDOC_LIBCMXA): $(LIBCMXFILES)
 manpages: stdlib_man/Pervasives.3o
 
 dot: $(EXECMOFILES)
-	$(OCAMLDOC_RUN) -dot -dot-reduce -o ocamldoc.dot $(INCLUDES) \
-	odoc*.ml
+	$(OCAMLDOC) -dot -dot-reduce -o ocamldoc.dot $(INCLUDES) odoc*.ml
 
 # Parsers and lexers dependencies :
 ###################################
@@ -295,37 +292,37 @@ installopt_really:
 ###########
 test: dummy
 	$(MKDIR) $@
-	$(OCAMLDOC_RUN) -html -colorize-code -sort -d $@ $(INCLUDES) -dump $@/ocamldoc.odoc odoc*.ml odoc*.mli -v
+	$(OCAMLDOC) -html -colorize-code -sort -d $@ $(INCLUDES) -dump $@/ocamldoc.odoc odoc*.ml odoc*.mli -v
 	$(MKDIR) $@-custom
-	$(OCAMLDOC_RUN) -colorize-code -sort -d $@-custom $(INCLUDES) \
+	$(OCAMLDOC) -colorize-code -sort -d $@-custom $(INCLUDES) \
 	-g generators/odoc_literate.cmo -g generators/odoc_todo.cmo \
 	-load $@/ocamldoc.odoc -v
 
 test_stdlib: dummy
 	$(MKDIR) $@
-	$(OCAMLDOC_RUN) -html -colorize-code -sort -d $@ $(INCLUDES) -dump $@/stdlib.odoc -keep-code \
+	$(OCAMLDOC) -html -colorize-code -sort -d $@ $(INCLUDES) -dump $@/stdlib.odoc -keep-code \
 	../stdlib/pervasives.ml ../stdlib/*.mli \
 	../otherlibs/$(UNIXLIB)/unix.mli \
 	../otherlibs/str/str.mli
 
 test_stdlib_code: dummy
 	$(MKDIR) $@
-	$(OCAMLDOC_RUN) -html -colorize-code -sort -d $@ $(INCLUDES) -dump $@/stdlib.odoc -keep-code \
+	$(OCAMLDOC) -html -colorize-code -sort -d $@ $(INCLUDES) -dump $@/stdlib.odoc -keep-code \
 	`ls ../stdlib/*.ml | grep -v Labels` \
 	../otherlibs/$(UNIXLIB)/unix.ml \
 	../otherlibs/str/str.ml
 
 test_framed: dummy
 	$(MKDIR) $@
-	$(OCAMLDOC_RUN) -g odoc_fhtml.cmo -sort -colorize-code -d $@ $(INCLUDES) odoc*.ml odoc*.mli
+	$(OCAMLDOC) -g odoc_fhtml.cmo -sort -colorize-code -d $@ $(INCLUDES) odoc*.ml odoc*.mli
 
 test_latex: dummy
 	$(MKDIR) $@
-	$(OCAMLDOC_RUN) -latex -sort -o $@/test.tex -d $@ $(INCLUDES) odoc*.ml odoc*.mli test2.txt ../stdlib/*.mli ../otherlibs/unix/unix.mli
+	$(OCAMLDOC) -latex -sort -o $@/test.tex -d $@ $(INCLUDES) odoc*.ml odoc*.mli test2.txt ../stdlib/*.mli ../otherlibs/unix/unix.mli
 
 test_latex_simple: dummy
 	$(MKDIR) $@
-	$(OCAMLDOC_RUN) -latex -sort -o $@/test.tex -d $@ $(INCLUDES) \
+	$(OCAMLDOC) -latex -sort -o $@/test.tex -d $@ $(INCLUDES) \
 	-latextitle 6,subsection -latextitle 7,subsubection \
 	../stdlib/hashtbl.mli \
 	../stdlib/arg.mli \
@@ -334,21 +331,21 @@ test_latex_simple: dummy
 
 test_man: dummy
 	$(MKDIR) $@
-	$(OCAMLDOC_RUN) -man -sort -d $@ $(INCLUDES) odoc*.ml odoc*.mli
+	$(OCAMLDOC) -man -sort -d $@ $(INCLUDES) odoc*.ml odoc*.mli
 
 test_texi: dummy
 	$(MKDIR) $@
-	$(OCAMLDOC_RUN) -texi -sort -d $@ $(INCLUDES) odoc*.ml odoc*.mli
+	$(OCAMLDOC) -texi -sort -d $@ $(INCLUDES) odoc*.ml odoc*.mli
 
 stdlib_man/Pervasives.3o: $(STDLIB_MLIS)
 	$(MKDIR) stdlib_man
-	$(OCAMLDOC_RUN) -man -d stdlib_man $(INCLUDES) \
+	$(OCAMLDOC) -man -d stdlib_man $(INCLUDES) \
 	-t "OCaml library" -man-mini \
 	$(STDLIB_MLIS)
 
 autotest_stdlib: dummy
 	$(MKDIR) $@
-	$(OCAMLDOC_RUN) -g autotest/odoc_test.cmo\
+	$(OCAMLDOC) -g autotest/odoc_test.cmo \
 	$(INCLUDES) -keep-code \
 	../stdlib/pervasives.ml ../stdlib/*.mli \
 	../otherlibs/$(UNIXLIB)/unix.mli \
-- 
1.8.1.4

