From 9f71bf994f1033410e970bc5a3ce68c9949525ac Mon Sep 17 00:00:00 2001
From: Adrien Nader <adrien@notk.org>
Date: Thu, 28 Feb 2013 23:02:57 +0100
Subject: [PATCH 32/34] fixup! 4af54559a53e150ec07679b8712394cdfddf878c

---
 build/partial-install.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/build/partial-install.sh b/build/partial-install.sh
index 465fffd..e332866 100755
--- a/build/partial-install.sh
+++ b/build/partial-install.sh
@@ -53,6 +53,7 @@ installbestbin() {
     else
       wontinstall "$1"
       wontinstall "$2"
+      return
     fi
   fi
   [ -x "$3" ] || chmod +x "$3"
-- 
1.8.1.4

