From 2733fb7c88a398d9d886256b1bcd9c6cf2df5257 Mon Sep 17 00:00:00 2001
From: Adrien Nader <adrien@notk.org>
Date: Thu, 28 Feb 2013 23:35:44 +0100
Subject: [PATCH 34/34] fixup! 4af54559a53e150ec07679b8712394cdfddf878c

---
 build/partial-install.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/build/partial-install.sh b/build/partial-install.sh
index a5f517a..514156f 100755
--- a/build/partial-install.sh
+++ b/build/partial-install.sh
@@ -182,6 +182,6 @@ installdir \
   ocamlbuild/ocamlbuild/ocamlbuild.$O \
   ocamlbuild/$LIBDIR/ocamlbuild
 
-inocamlbuild/stalldir \
-  ocamlbuild/../ocamlbuild/man/ocamlbuild.1 \
-  ocamlbuild/$MANDIR/man1
+installdir \
+  ../ocamlbuild/man/ocamlbuild.1 \
+  $MANDIR/man1
-- 
1.8.1.4

